src/tools/etherscanClient.ts

```diff
@@ -614,33 +629,145 @@ export class EtherscanClient {
   }
 
   public async isBlockIndexedAsync(chainId: number, blockNumber: number): Promise<boolean> {
```

Comment:
This method is used to check whether a given block has been indexed by Etherscan, instead of if it has been mined by the blockchain network. Etherscan will index mined blocks, generally with a time delay. Although normally that delay is within a few seconds, when Etherscan service is in maintenance mode, and have any incidents, the delay could be hours or even days.
So checking the latest mined block is incorrect and we need to use Etherscan API to tell if they have indexed a block. 
For chains like Plasma where the `getblockreward` method is not available, simply return true while print a warning method.

Reply:
For Plasma where we have no reliable way to check if a block is indexed, it is good to use `latestBlock - safetyBuffer`. For other chains, let's use the original method. 

So the getLatestBlockNumber() method is for internal use only. It is good to use plasmascan's proxy module to avoid introduce additional dependency. It is also good to keep this `getLatestBlockNumber()` method private.

Reply:
@thomasle-vn let's make this more clear: **For other chains, let's use the original method.** Only on Plasma where we have no reliable way to check if a block is indexed, it is good to use latestBlock - safetyBuffer

Reply:
@thomasle-vn Could you make this change? Thanks!